### ISSUE TITLE
Change JsonTypeName in DefaultingFileAppenderFactory

### ISSUE THREAD

**Issue by sleberknight** [sleberknight/kiwi-beta#286]
Some of our own services already use a custom `FileAppenderFactory` which has a `@JsonTypeName` of `rolling`. Because of this, simply adding `kiwi-beta` as a dependency of that service causes the `DefaultingFileAppenderFactory` to override the custom one when the `META-INF/services/io.dropwizard.logging.AppenderFactory` files are combined using the Maven Shade Plugin to build the "fat service jar".

In other words, there are _two_ implementations of `FileAppenderFactory ` both with the same `@JsonTypeName`. This causes confusion. To avoid this, rename the `@JsonTypeName` in `DefaultingFileAppenderFactory ` to something different.

I categorized this an an _enhancement_ instead of a bug, since it technically is _not_ a bug. Rather, it can cause a conflict and the "wrong" implementation to be used.
